contracts/SpokePool.sol

```diff
@@ -667,6 +681,81 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RequestRefund event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would in the worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @dev Caller needs to pass in `fillBlock` that the FilledRelay event was emitted on the `destinationChainId`.
+     * This is to make it hard to request a refund before a fill has been mined and to make lookups of the original
+     * fill as simple as possible.
+     * @param relayer Relayer on FilledRelay event on destinationChain we want to match with this Refund.
+     * This should be account receiving any refund stemming from this event.
+     * @param destinationToken This chain's destination token equivalent for original deposit destination token.
+     * @param amount Original deposit amount.
+     * @param originChainId Original origin chain ID.
+     * @param realizedLpFeePct Original realized LP fee %.
+     * @param depositId Original deposit ID.
+     * @param maxCount Max count to protect the refund recipient from frontrunning.
+     */
+    function requestRefund(
+        address relayer,
```

Comment:
What was the thinking around having the relayer address as an argument to this function, rather than just using msg.sender?

As is, it seems that this allows anyone to arbitrarily request a refund on behalf of another relayer. I.e. it could be probably be used to the advantage of a prospective depositor on the same chain by bundling a refund request for _another_ relayer before their own deposit in order to maximise their benefit/minimise their penalty.

Maybe this is actually desirable - i.e. it's another case of pushing the risk onto the relayer, and it provides others with additional ways of keeping the SpokePool at its target balance.

Reply:
Do you think we should require that only the `relayer` can call `requestRefund`? This is an interesting point and I think it has pros and cons so maybe we just remove the edge case entirely and `require(msg.sender == relayer)`

Reply:
> Do you think we should require that only the `relayer` can call `requestRefund`?

The comments give me the impression that we assume that a relayer will only ever self-request, so I did get curious about the other scenarios. I was actually wondering about the case where a relayer might fill and hold onto its repayment request until the SpokePool on the repayment chain was above its target balance. In this case there could theoretically be many bundles between making a fill and requesting a refund...though if it's permissionless to request a refund on behalf of another relayer then that relayer would probably be more interested in requesting its refund as soon as the fill is mined (and the block number has a low probability of a re-org).

> This is an interesting point and I think it has pros and cons so maybe we just remove the edge case entirely and `require(msg.sender == relayer)`

That would definitely simplify the reasoning. In that case, would it just be better to remove the relayer argument to the requestRefund() function?

Reply:
Hm these are interesting examples. Requiring msg.sender == relayer would open up possibility that relayers waits a long time to request a refund. Although it would also be annoying to build a bot that goes through all Fills and requests refunds for them. Maybe we should build this sort of bot?

Reply:
Maybe we can enforce in the UMIP that the refund request is between -2 and +2 hours of the Fill based on a "reasonable" computation of the Fill time. This would be a bit hard to enforce in court but the spirit of the law would protect the system but opoortunistic relayers

Reply:
Had a good discussion with @mrice32 about this, its actually not a problem if the relayer waits until its profitable for them to execute the refund. That's exactly what the repayment chain SpokePool wants--to have relayers withdraw funds when its over balance and hold off when its under balance.

The change is implemented here not as a `require` statement, since we don't know the `relayer` address from the destination chain, so its up to the off chain validator to check this

Reply:
@nicholaspai 100% agree!